src/map/figure.c

```diff
@@ -14,6 +14,13 @@ int map_figure_at(int grid_offset)
     return map_grid_is_valid_offset(grid_offset) ? figures.items[grid_offset] : 0;
 }
 
+static void cap_figures_on_same_tile_index(figure *f)
```

Comment:
Suggestion: instead of capping the value at 20, let it go as high as it wants, then on `adjust_pixel_offset`, for the array index, use `f->figures_on_same_tile_index % 20`.

This has the advantage that it will also work on other eventual spots that also cause the issue.

It will also "centralise" the limit of figures per tile, avoiding the magic number of `20` here.

Reply:
I was trying to maintain the original logic as much as possible since this is a Julius change. @bvschaik are you ok with @crudelios suggestion?